archunit/src/test/java/com/tngtech/archunit/lang/ArchRuleTest.java

```diff
@@ -159,6 +161,20 @@ public void finish(ConditionEvents events) {
         assertThat(condition.eventsFromFinish.getViolating()).hasSize(1);
     }
 
+    @Test
+    public void empty_input_throws_an_assertion_exception_by_default() {
+        thrown.expect(AssertionError.class);
+
+        classes().should().haveSimpleName("NoClassesShouldBePassedAsInput").evaluate(importClassesWithContext());
```

Comment:
You could create the opposite of `com.tngtech.archunit.lang.ArchRuleTest#ALWAYS_BE_VIOLATED`. I.e. a condition that never causes any error by just being empty. Then it's clear that this rule can never fail in a regular way. Also, I would at least once test that the filtering to 0 classes breaks the rule and not only an empty input set (compare my comment from above about the rule process). It's an important aspect that a non-empty set of classes that is filtered down to 0 classes also causes this failure.

Reply:
Created `ALWAYS_BE_VALID` and added a test, where a non-empty set of classes is filtered down to 0.